utility/utility.py

```diff
@@ -11,9 +12,41 @@ def getRole(roles, roleName):
     return None
 
 
-# restricts roles that can be picked based on roles.py
-def isGamesRole(role):
-    for validRole in ROLES_LIST:
-        if validRole.name.casefold() == str(role).casefold():
+# find bot's member object in guild
+# returns None
+def getBotMember(guild, bot):
+    for m in guild.members:
+        if m == bot.user:
+            return m
+    return None
+
+
+# returns if role is found in guild, has no permissions
+# and is manageable by the bot
+def isGamesRole(guild, bot, role):
+    botMember = getBotMember(guild, bot)
+
+    if botMember is None:
+        return False
+
+    # get the highest role with manage_role permission
+    botRole = None
+    manageRolePermission = Permissions.none()
+    manageRolePermission.manage_roles = True
+    for r in botMember.roles:
```

Comment:
Are roles ordered so that the roles with least permissions are first?

Reply:
Roles are ordered from 0 to n, n being the highest priority role. When we add roles, they're always places at 0 and all other roles are bumped up.

I'm not sure if I get what you're saying, but I'll try to explain my theory. I'm finding the highest priority role of bot that allows it to manage role. I think it's up to the server managers to handle the bots permissions. Admin naturally comes with manage role and if it somehow gets put on the bot, Lord have mercy.

Reply:
Are these checks necessary? According to [this part of the Discord API documentation](https://discord.com/developers/docs/topics/permissions#permission-hierarchy), a bot can only grant roles that are lower permission than its highest role. If that's correct, the last two checks here should already be blocked by the API.

Reply:
Good point. I'll eliminate some redundant checks. I was trying to make illustrative error feedback, but I can use except to catch permission errors instead.